test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_create.cpp

```diff
@@ -289,8 +289,9 @@ struct CreateCommandBufferDeviceDoesNotSupportOutOfOderQueue
     {
         BasicCommandBufferTest::Skip();
 
-        // If device supports out of order queues test should be skipped
-        return out_of_order_support != 0;
+        // If device does not support out of order queue or if device supports
+        // out of order command buffer test should be skipped
+        return queue_out_of_order_support && !out_of_order_support;
```

Comment:
This test is a negative test to verify that the correct error is returned when a device doesn't have `out_of_order_support` command-buffer support, but it is attempted to be create one with an out-of-order queue. So I agree we do need the `queue_out_of_order_support` check you've added and skip if a device can't create an out-or-order queue to use for the test.

However I'm not quite following the current logic why we only skip the test if a device has out-of-order queue support and another condition. I think the check could be written as follows to better reflect the updated code comment.
```cpp
return !queue_out_of_order_support || (out_of_order_support !=0)
```